server/src/main/java/org/apache/iotdb/db/metadata/mnode/MeasurementMNode.java

```diff
@@ -153,42 +154,20 @@ public void setTriggerExecutor(TriggerExecutor triggerExecutor) {
   }
 
   @Override
-  public TimeValuePair getCachedLast() {
-    return cachedLastValuePair;
-  }
-
-  /**
-   * update last point cache
-   *
-   * @param timeValuePair last point
-   * @param highPriorityUpdate whether it's a high priority update
-   * @param latestFlushedTime latest flushed time
-   */
-  @Override
-  public synchronized void updateCachedLast(
-      TimeValuePair timeValuePair, boolean highPriorityUpdate, Long latestFlushedTime) {
-    if (timeValuePair == null || timeValuePair.getValue() == null) {
-      return;
-    }
-
-    if (cachedLastValuePair == null) {
-      // If no cached last, (1) a last query (2) an unseq insertion or (3) a seq insertion will
-      // update cache.
-      if (!highPriorityUpdate || latestFlushedTime <= timeValuePair.getTimestamp()) {
-        cachedLastValuePair =
-            new TimeValuePair(timeValuePair.getTimestamp(), timeValuePair.getValue());
+  public ILastCacheEntry getLastCacheEntry() {
+    if (lastCacheEntry == null) {
+      synchronized (this) {
+        if (lastCacheEntry == null) {
+          lastCacheEntry = new LastCacheEntry();
+        }
```

Comment:
It's good~ Veryyyy safe :D

Reply:
I think you should make `lastCacheEntry` a `volitale` field if you really want to avoid repeated initialization. Otherwise, such an implementation would actually be problematic for concurrent accessing.

Reply:
Actually, I think it's fine even if we don't take any control on this initailization...

Reply:
To avoid multi threads creating different lastCacheEntries on one MeasurementMNode, such control is necessary, otherwise entry merge should be considered. Volatile is important as well. Thanks for suggestions.

Reply:
Fixed